server/widget.c

```diff
@@ -85,7 +86,7 @@ widget_create(char *id, WidgetType type, Screen *screen)
 	debug(RPT_DEBUG, "%s(id=\"%s\", type=%d, screen=[%s])", __FUNCTION__, id, type, screen->id);
 
 	/* Create it */
-	w = malloc(sizeof(Widget));
+	w = calloc(1, sizeof(Widget));
```

Comment:
Otherwise the new begin_label, end_label and promille fields explicitly set to NULL/0. I'm used to just using calloc so everything in the struct has a known value (NULL/0) and only when a different value is used explicitly initialize things. If you prefer I can add explicit initialization of the 3 new members.

Reply:
It's ok. I was just confused, because it looked like an unrelated change.